src/xtgeo/surface/_regsurf_ijxyz_parser.py

```diff
@@ -0,0 +1,267 @@
+34	3	775103.586551	6735995.770364	1687.320435
+...
+272	3	779791.271455	6735169.205039	1404.342041
+274	3	779830.663765	6735162.259112	1413.154175
+24	6	774915.307409	6736079.740388	1727.455078
+26	6	774954.699719	6736072.794461	1719.031128
+28	6	774994.092029	6736065.848533	1710.606201
+30	6	775033.484339	6736058.902606	1702.161255
+32	6	775072.876649	6736051.956679	1693.707153
+34	6	775112.268959	6736045.010752	1685.164673
+
+Here every second inline and every third xline is applied
+
+========================================================================================
+
+ 1312  2015   655905.938642   6627044.343360   1671.420800
+ 1312  2010   655858.839186   6627003.259513   1669.091400
+ 1302  2028   656151.649337   6627009.862811   1680.437500
+ 1312  2011   655868.259077   6627011.476282   1669.444000
+ 1302  2029   656161.069228   6627018.079581   1681.000000
+ 1302  2026   656132.809555   6626993.429272   1679.312500
+ 1302  2027   656142.229446   6627001.646042   1679.875000
+ 1302  2024   656113.969773   6626976.995733   1678.187500
+ 1302  2025   656123.389664   6626985.212503   1678.750000
+
+This seems to lack obvious patterns (mostly due to many undefied cells?). Hence we
+need to detect minimum spacing and min/max of both ilines and xlines, unless a template
+is applied.
+"""
+from __future__ import annotations
+
+from dataclasses import dataclass, field
+from typing import Literal, Optional
+
+import numpy as np
+from numpy.ma import MaskedArray
+
+import xtgeo
+from xtgeo.common.calc import find_flip, xyori_from_ij
+from xtgeo.common.log import null_logger
+
+logger = null_logger(__name__)
+
+
+@dataclass
+class SurfaceIJXYZ:
+    """Temporary class for parsing IJXYZ settings.
+
+    Attributes:
+        xcoords_in: Input X coordinates.
+        ycoords_in: Input Y coordinates.
+        values_in: Input values.
+        ilines_in: Input inline values.
+        xlines_in: Input crossline values.
+        template: Geometrical template.
+    """
+
+    xcoords_in: MaskedArray
+    ycoords_in: MaskedArray
+    values_in: MaskedArray
+    ilines_in: np.ndarray
+    xlines_in: np.ndarray
+    template: Optional[xtgeo.RegularSurface | xtgeo.Cube] = None
+
+    ncol: int = field(default=1, init=False)
+    nrow: int = field(default=1, init=False)
+    xori: float = field(default=0.0, init=False)
+    yori: float = field(default=0.0, init=False)
+    xinc: float = field(default=1.0, init=False)
+    yinc: float = field(default=1.0, init=False)
+    yflip: Literal[-1, 1] = field(default=1, init=False)
+    xcoords: MaskedArray = field(default_factory=MaskedArray, init=False)
+    ycoords: MaskedArray = field(default_factory=MaskedArray, init=False)
+    values: MaskedArray = field(default_factory=MaskedArray, init=False)
+    ilines: np.ndarray = field(default_factory=lambda: np.zeros(1), init=False)
+    xlines: np.ndarray = field(default_factory=lambda: np.zeros(1), init=False)
+
+    def __post_init__(self):
```

Comment:
```suggestion
    def __post_init__(self) -> None:
```

One little annotation here too 😄